### PR TITLE
Feat: Adapt latest DID implementations

### LINKED ISSUE #116: [Task]: Sync v2 branch with dev
## Task description

We have to merge develop back into v2 and adapt the latest DID implementations.

### PR DESCRIPTION
**Pull request by rubalv96**
# Description of change

This PR aims to solve the second part of #116  (adapt the latest DID implementations).
This PR closes #116

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [X] My code follows the contribution guidelines for this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
When you introduce a wrong DID there is no error message like in https://explorer.iota.org/mainnet/identity-resolver/did:iota:3TT7QsmESw1dcboV2oVTAuJbbxaVYAThexZFct2z5kkk

File: client/src/app/routes/IdentityResolver.tsx [hunk 1]
```diff
@@ -190,81 +190,84 @@ class IdentityResolver extends AsyncComponent<
...
-                                                                target="_blank"
-                                                                rel="noreferrer"
-                                                            >
-                                                                latest version
-                                                            </a>.
-                                                        </div>
-                                                    )}
-
-                                                    {!this.state.error &&
-                                                        !(this.state.latestMessageId === this.EMPTY_MESSAGE_ID) && (
```
Comment (begonaalvarezd) [inline client/src/app/routes/IdentityResolver.tsx:218]: this chunk of code is missing
```
{this.state.error && (
    <div className="identity-json-container did-error">
        <p className="margin-b-t">ಠ_ಠ </p>
        <p className="">{this.state.errorMessage}</p>
    </div>
)}
```

File: client/src/app/routes/IdentityResolver.tsx [hunk 2]
```diff
@@ -190,81 +190,84 @@ class IdentityResolver extends AsyncComponent<
...
+                                                            library. If this is your DID you can recreate it using the
+                                                            {" "}
+                                                        </span>
+                                                        <a
+                                                            href="https://github.com/iotaledger/identity.rs/releases"
+                                                            target="_blank"
+                                                            rel="noreferrer"
+                                                        >
+                                                            latest version
+                                                        </a>.
```
Comment (brancoder) [inline client/src/app/routes/IdentityResolver.tsx:209]: Maybe we can change the color of the a-tag to inform the user that it's a link.

image

Comment (begonaalvarezd) [reply client/src/app/routes/IdentityResolver.tsx:209]: pushed

**Review by begonaalvarezd — APPROVED** [review]

**Review by brancoder — APPROVED** [review]

**Pull request merged**